sdk/identity/azure-identity/azure/identity/_credentials/linux_vscode_adapter.py

```diff
@@ -0,0 +1,48 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import os
+import json
+import ctypes as ct
+from .._constants import VSCODE_CREDENTIALS_SECTION
+
+
+def _c_str(string):
+    return ct.c_char_p(string.encode('utf-8'))
+
+
+_libsecret = ct.cdll.LoadLibrary('libsecret-1.so.0')
+_libsecret.secret_schema_new.argtypes = \
+    [ct.c_char_p, ct.c_uint, ct.c_char_p, ct.c_uint, ct.c_char_p, ct.c_uint, ct.c_void_p]
+_libsecret.secret_password_lookup_sync.argtypes = \
+    [ct.c_void_p, ct.c_void_p, ct.c_void_p, ct.c_char_p, ct.c_char_p, ct.c_char_p, ct.c_char_p, ct.c_void_p]
+
+
+def _get_user_settings_path():
+    app_data_folder = os.environ['HOME']
+    return os.path.join(app_data_folder, ".config", "Code", "User", "settings.json")
+
+
+def _get_user_settings():
+    path = _get_user_settings_path()
+    try:
+        with open(path) as file:
+            data = json.load(file)
+            environment_name = data.get("azure.cloud", "Azure")
+            return environment_name
+    except IOError:
+        return "Azure"
+
+
+def _get_refresh_token(service_name, account_name):
+    schema = _libsecret.secret_schema_new(_c_str("org.freedesktop.Secret.Generic"), 2,
+                                          _c_str("service"), 0, _c_str("account"), 0, None)
+    return _libsecret.secret_password_lookup_sync(schema, None, None, _c_str("service"), _c_str(service_name),
```

Comment:
Should we anticipate exceptions here? Could this return a truthy sentinel value instead of a refresh token? I'm curious what happens when:
- libsecret isn't available
- libsecret is available but X11 is not (e.g. in an SSH session)
- the secret isn't found